kernel/system_work_q.c

```diff
@@ -26,10 +26,12 @@ static int k_sys_work_q_init(void)
 		.no_yield = IS_ENABLED(CONFIG_SYSTEM_WORKQUEUE_NO_YIELD),
 	};
 
+	k_sys_work_q.thread.base.user_options |= K_ESSENTIAL;
 	k_work_queue_start(&k_sys_work_q,
```

Comment:
Your suggestion of adding it to the config probably makes the most sense - I should have clarified earlier